apps/web/src/common/components/FormController/index.tsx

```diff
@@ -0,0 +1,147 @@
+                  fullWidth={fullWidth}
+                  label={label}
+                  inputRef={ref}
+                  {...field}
+                  error={props.fieldState.invalid}
+                  helperText={props.fieldState.error?.message}
+                  onChange={(event) =>
+                    field.onChange(
+                      event.target.value === '' ? '' : +event.target.value,
+                    )
+                  }
+                  type="number"
+                  {...inputProps}
+                />
+              )
+            if (type === 'uploadFile')
+              return (
+                <UploadFile
+                  label={label}
+                  handleSelectFile={handleUploadFile!}
+                  error={props.fieldState.invalid}
+                  errorMessage={props.fieldState.error?.message}
+                  url={field.value}
+                  {...inputProps}
+                />
+              )
+
+            if (type === 'select')
+              return (
+                <TextField
+                  select
+                  required={!optional}
+                  fullWidth={fullWidth}
+                  label={label}
+                  inputRef={ref}
+                  {...field}
+                  error={props.fieldState.invalid}
+                  helperText={props.fieldState.error?.message}
+                  {...inputProps}
+                >
+                  {selectProps?.map((choice) => (
+                    <MenuItem
+                      {...choice}
+                      key={choice.value?.toString() || `${Math.random()}`}
+                    />
+                  ))}
+                </TextField>
+              )
+            if (type === 'date')
+              return (
+                <DesktopDatePicker
+                  label={label}
+                  inputFormat="DD/MM/YYYY"
+                  inputRef={ref}
+                  {...field}
+                  onChange={(newValue) => field.onChange(newValue)}
+                  renderInput={(params) => <TextField {...params} fullWidth />}
+                />
+              )
+            if (type === 'time')
+              return (
+                <TimePicker
+                  label={label}
+                  inputRef={ref}
+                  {...field}
+                  onChange={(newValue) => field.onChange(newValue)}
+                  renderInput={(params) => <TextField {...params} fullWidth />}
+                />
+              )
+
+            return (
+              <PasswordTextField
+                required={!optional}
+                fullWidth={fullWidth}
+                label={label}
+                {...field}
+                inputRef={ref}
+                error={props.fieldState.invalid}
+                helperText={props.fieldState.error?.message}
+                {...inputProps}
+              />
+            )
+          }}
+        />
+      )}
+    </Grid>
```

Comment:
(Discussion)
Can we refactor this to another `useMemo` but I'm not sure that this will cause some impact to performance or not

Reply:
this might be a good idea (some company start with "L" do something similar to this :joy:)